internal/provider/resource_secret_test.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the Apache License, Version 2.0, see LICENCE file for details.
+
+package provider
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	internaltesting "github.com/juju/terraform-provider-juju/internal/testing"
+)
+
+func TestAcc_ResourceSecret_CreateWithInfo(t *testing.T) {
+	modelName := acctest.RandomWithPrefix("tf-test-model")
+	secretName := "tf-test-secret"
+	secretInfo := "test-info"
+	secretValue := map[string]string{
+		"key1": "value1",
+		"key2": "value2",
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: frameworkProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccResourceSecret(modelName, secretName, secretValue, secretInfo),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("juju_secret."+secretName, "model", modelName),
+					resource.TestCheckResourceAttr("juju_secret."+secretName, "name", secretName),
+					resource.TestCheckResourceAttr("juju_secret."+secretName, "info", secretInfo),
+					resource.TestCheckResourceAttr("juju_secret."+secretName, "value.key1", "value1"),
+					resource.TestCheckResourceAttr("juju_secret."+secretName, "value.key2", "value2"),
+				),
+			},
+		},
+	})
+}
+
+func TestAcc_ResourceSecret_CreateWithNoInfo(t *testing.T) {
+	modelName := acctest.RandomWithPrefix("tf-test-model")
+	secretName := "tf-test-secret"
+	secretValue := map[string]string{
+		"key1": "value1",
+		"key2": "value2",
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: frameworkProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccResourceSecret(modelName, secretName, secretValue, ""),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("juju_secret."+secretName, "model", modelName),
+					resource.TestCheckResourceAttr("juju_secret."+secretName, "name", secretName),
+					resource.TestCheckResourceAttr("juju_secret."+secretName, "value.key1", "value1"),
+					resource.TestCheckResourceAttr("juju_secret."+secretName, "value.key2", "value2"),
+				),
+			},
+		},
+	})
+}
```

Comment:
If name is optional there should probably also be a test with no name.

Reply:
added